crates/bin/node/src/response_cache.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    fmt::Debug,
+    time::{Duration, Instant},
+};
+
+use crate::errors;
+use dashmap::DashMap;
+use node::{MeltResponse, MintResponse, SwapResponse};
+
+pub trait ResponseCache<K, V> {
+    // Basic operations
+    fn get(&self, key: &K) -> Option<V>;
+    fn insert(&self, key: K, value: V) -> Result<(), errors::Error>;
+    fn remove(&self, key: &K) -> bool;
+
+    // For cleanup/shutdown
```

Comment:
I don't think we ever want to cleanup the whole cache.
Even on shutdown, we may instead want to prune it, and then store the remaining cached values on the pgdb and load them back into the cache on the next launch. Don't you think?